csharp/extractor/Semmle.Extraction.CIL/Entities/NoMetadataHandleType.FullyQualifiedNameParser.cs

```diff
@@ -0,0 +1,174 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Semmle.Extraction.CIL.Entities
+{
+    internal sealed partial class NoMetadataHandleType
+    {
+        /// <summary>
+        /// Parser to split a fully qualified name into short name, namespace or declaring type name, assembly name, and
+        /// type argument names. Names are in the following format:
+        /// <c>N1.N2.T1`2+T2`2[T3,[T4, A1, Version=...],T5,T6], A2, Version=...</c>
+        /// </summary>
+        /// <example>
+        /// <code>typeof(System.Collections.Generic.List<int>.Enumerator)
+        /// -> System.Collections.Generic.List`1+Enumerator[[System.Int32, System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e
+        /// typeof(System.Collections.Generic.List<>.Enumerator)
+        /// -> System.Collections.Generic.List`1+Enumerator, System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e
+        /// </code>
+        /// </example>
+        private class FullyQualifiedNameParser
+        {
+            public string ShortName { get; internal set; }
+            public string? AssemblyName { get; internal set; }
+            public IEnumerable<string>? TypeArguments { get; internal set; }
+            public string? UnboundGenericTypeName { get; internal set; }
+            public string ContainerName { get; internal set; }
+            public bool IsContainerNamespace { get; internal set; }
+
+            private string AssemblySuffix => string.IsNullOrWhiteSpace(AssemblyName) ? "" : $", {AssemblyName}";
+
+            public FullyQualifiedNameParser(string name)
+            {
+                ExtractAssemblyName(ref name, out var lastBracketIndex);
+                ExtractTypeArguments(ref name, lastBracketIndex, out var containerTypeArguments);
+                ExtractContainer(ref name, containerTypeArguments);
```

Comment:
Consider adding an `out string containerName` to this call, and assign the value to `ContainerName` to avoid the nullability warning.